### ISSUE TITLE
prom_remote_write failed from v1.30.0-to v1.34.0

### ISSUE THREAD

**Issue by saintak** [GuanceCloud/datakit#24]
Impact version range：1.30.0 - 1.34.0
The last normal version: 1.29.1
Env:
- I configured it in Prom:
```yaml
remote_write:
 - url: "http://127.0.0.1:9529/prom_remote_write"
```
- I have enabled prom_demote_frite.conf in the datakit
Problem:
- when I use the version 1.29.1 everything is ok
- When I use versions 1.30.0 and later, the datakit monitor shows normal prom_remote_write input, but the console.guance.com cannot see the data uploaded by prom.However, other non Prom data can be uploaded normally. So I think that since this version, there have been issues with the function, and manually downgrading to 1.29.1 can return normal 
- there is no errorlog
```
log:2024-08-05T15:19:42.300+0800	INFO	config	config/load_inputconf.go:162	find 13 confs: []string{"/usr/local/datakit/conf.d/container/container.conf", "/usr/local/datakit/conf.d/datakit.conf", "/usr/local/datakit/conf.d/host/cpu.conf", "/usr/local/datakit/conf.d/host/disk.conf", "/usr/local/datakit/conf.d/host/diskio.conf", "/usr/local/datakit/conf.d/host/dk.conf", "/usr/local/datakit/conf.d/host/host_processes.conf", "/usr/local/datakit/conf.d/host/hostobject.conf", "/usr/local/datakit/conf.d/host/mem.conf", "/usr/local/datakit/conf.d/host/net.conf", "/usr/local/datakit/conf.d/host/swap.conf", "/usr/local/datakit/conf.d/host/system.conf", "/usr/local/datakit/conf.d/prom/prom_remote_write.conf"}
log:2024-08-05T15:19:42.303+0800	INFO	config	config/inputcfg.go:143	load input "prom_remote_write" from conf file
log:2024-08-05T15:19:42.306+0800	INFO	http	httpapi/reg.go:64	register PUT@/prom_remote_write to HTTP server
log:2024-08-05T15:19:42.306+0800	INFO	http	httpapi/reg.go:64	register POST@/prom_remote_write to HTTP server
log:2024-08-05T15:19:42.307+0800	INFO	usage_trace	usagetrace/trace.go:139	add reserved inputs []string{"rum", "kafkamq", "prom_remote_write", "beats_output"}
log:2024-08-05T15:19:42.307+0800	INFO	usage_trace	usagetrace/trace.go:152	add reserved input "prom_remote_write"
log:2024-08-05T15:19:51.242+0800	INFO	inputs	inputs/inputs.go:361	starting input prom_remote_write ...
log:2024-08-05T15:19:51.242+0800	INFO	prom_remote_write	promremote/input.go:94	prom_remote_write input started...
log:2024-08-05T15:19:51.242+0800	INFO	inputs	inputs/inputs.go:365	input prom_remote_write exited, this maybe a input that only register a HTTP handle
log:2024-08-05T15:44:47.615+0800	INFO	config	config/load_inputconf.go:162	find 13 confs: []string{"/usr/local/datakit/conf.d/container/container.conf", "/usr/local/datakit/conf.d/datakit.conf", "/usr/local/datakit/conf.d/host/cpu.conf", "/usr/local/datakit/conf.d/host/disk.conf", "/usr/local/datakit/conf.d/host/diskio.conf", "/usr/local/datakit/conf.d/host/dk.conf", "/usr/local/datakit/conf.d/host/host_processes.conf", "/usr/local/datakit/conf.d/host/hostobject.conf", "/usr/local/datakit/conf.d/host/mem.conf", "/usr/local/datakit/conf.d/host/net.conf", "/usr/local/datakit/conf.d/host/swap.conf", "/usr/local/datakit/conf.d/host/system.conf", "/usr/local/datakit/conf.d/prom/prom_remote_write.conf"}
log:2024-08-05T15:44:47.617+0800	INFO	config	config/inputcfg.go:143	load input "prom_remote_write" from conf file
log:2024-08-05T15:44:47.620+0800	INFO	usage_trace	usagetrace/trace.go:139	add reserved inputs []string{"rum", "kafkamq", "prom_remote_write", "beats_output"}
log:2024-08-05T15:44:47.620+0800	INFO	http	httpapi/reg.go:64	register POST@/prom_remote_write to HTTP server
log:2024-08-05T15:44:47.620+0800	INFO	http	httpapi/reg.go:64	register PUT@/prom_remote_write to HTTP server
log:2024-08-05T15:44:47.620+0800	INFO	usage_trace	usagetrace/trace.go:152	add reserved input "prom_remote_write"
log:2024-08-05T15:44:49.616+0800	INFO	inputs	inputs/inputs.go:361	starting input prom_remote_write ...
log:2024-08-05T15:44:49.616+0800	INFO	prom_remote_write	promremote/input.go:94	prom_remote_write input started...
log:2024-08-05T15:44:49.616+0800	INFO	inputs	inputs/inputs.go:365	input prom_remote_write exited, this maybe a input that only register a HTTP handle
log:2024-08-05T15:49:50.212+0800	INFO	config	config/load_inputconf.go:162	find 13 confs: []string{"/usr/local/datakit/conf.d/container/container.conf", "/usr/local/datakit/conf.d/datakit.conf", "/usr/local/datakit/conf.d/host/cpu.conf", "/usr/local/datakit/conf.d/host/disk.conf", "/usr/local/datakit/conf.d/host/diskio.conf", "/usr/local/datakit/conf.d/host/dk.conf", "/usr/local/datakit/conf.d/host/host_processes.conf", "/usr/local/datakit/conf.d/host/hostobject.conf", "/usr/local/datakit/conf.d/host/mem.conf", "/usr/local/datakit/conf.d/host/net.conf", "/usr/local/datakit/conf.d/host/swap.conf", "/usr/local/datakit/conf.d/host/system.conf", "/usr/local/datakit/conf.d/prom/prom_remote_write.conf"}
log:2024-08-05T15:49:50.215+0800	INFO	config	config/inputcfg.go:143	load input "prom_remote_write" from conf file
log:2024-08-05T15:49:50.217+0800	INFO	usage_trace	usagetrace/trace.go:139	add reserved inputs []string{"rum", "kafkamq", "prom_remote_write", "beats_output"}
log:2024-08-05T15:49:50.218+0800	INFO	http	httpapi/reg.go:64	register PUT@/prom_remote_write to HTTP server
log:2024-08-05T15:49:50.218+0800	INFO	http	httpapi/reg.go:64	register POST@/prom_remote_write to HTTP server
log:2024-08-05T15:49:50.218+0800	INFO	usage_trace	usagetrace/trace.go:152	add reserved input "prom_remote_write"
log:2024-08-05T15:49:51.977+0800	INFO	inputs	inputs/inputs.go:361	starting input prom_remote_write ...
log:2024-08-05T15:49:51.978+0800	INFO	prom_remote_write	promremote/input.go:94	prom_remote_write input started...
log:2024-08-05T15:49:51.978+0800	INFO	inputs	inputs/inputs.go:365	input prom_remote_write exited, this maybe a input that only register a HTTP handle
log:2024-08-05T15:50:23.835+0800	INFO	config	config/load_inputconf.go:162	find 13 confs: []string{"/usr/local/datakit/conf.d/container/container.conf", "/usr/local/datakit/conf.d/datakit.conf", "/usr/local/datakit/conf.d/host/cpu.conf", "/usr/local/datakit/conf.d/host/disk.conf", "/usr/local/datakit/conf.d/host/diskio.conf", "/usr/local/datakit/conf.d/host/dk.conf", "/usr/local/datakit/conf.d/host/host_processes.conf", "/usr/local/datakit/conf.d/host/hostobject.conf", "/usr/local/datakit/conf.d/host/mem.conf", "/usr/local/datakit/conf.d/host/net.conf", "/usr/local/datakit/conf.d/host/swap.conf", "/usr/local/datakit/conf.d/host/system.conf", "/usr/local/datakit/conf.d/prom/prom_remote_write.conf"}
log:2024-08-05T15:50:23.837+0800	INFO	config	config/inputcfg.go:143	load input "prom_remote_write" from conf file
log:2024-08-05T15:50:23.839+0800	INFO	http	httpapi/reg.go:64	register PUT@/prom_remote_write to HTTP server
log:2024-08-05T15:50:23.839+0800	INFO	http	httpapi/reg.go:64	register POST@/prom_remote_write to HTTP server
log:2024-08-05T15:50:23.840+0800	INFO	usage_trace	usagetrace/trace.go:139	add reserved inputs []string{"rum", "kafkamq", "prom_remote_write", "beats_output"}
log:2024-08-05T15:50:23.840+0800	INFO	usage_trace	usagetrace/trace.go:152	add reserved input "prom_remote_write"
log:2024-08-05T15:50:26.299+0800	INFO	inputs	inputs/inputs.go:361	starting input prom_remote_write ...
log:2024-08-05T15:50:26.299+0800	INFO	prom_remote_write	promremote/input.go:94	prom_remote_write input started...
log:2024-08-05T15:50:26.299+0800	INFO	inputs	inputs/inputs.go:365	input prom_remote_write exited, this maybe a input that only register a HTTP handle
log:2024-08-05T16:05:32.793+0800	INFO	config	config/load_inputconf.go:162	find 13 confs: []string{"/usr/local/datakit/conf.d/container/container.conf", "/usr/local/datakit/conf.d/datakit.conf", "/usr/local/datakit/conf.d/host/cpu.conf", "/usr/local/datakit/conf.d/host/disk.conf", "/usr/local/datakit/conf.d/host/diskio.conf", "/usr/local/datakit/conf.d/host/dk.conf", "/usr/local/datakit/conf.d/host/host_processes.conf", "/usr/local/datakit/conf.d/host/hostobject.conf", "/usr/local/datakit/conf.d/host/mem.conf", "/usr/local/datakit/conf.d/host/net.conf", "/usr/local/datakit/conf.d/host/swap.conf", "/usr/local/datakit/conf.d/host/system.conf", "/usr/local/datakit/conf.d/prom/prom_remote_write.conf"}
log:2024-08-05T16:05:32.795+0800	INFO	config	config/inputcfg.go:143	load input "prom_remote_write" from conf file
log:2024-08-05T16:05:32.798+0800	INFO	usage_trace	usagetrace/trace.go:139	add reserved inputs []string{"rum", "kafkamq", "prom_remote_write", "beats_output"}
log:2024-08-05T16:05:32.798+0800	INFO	http	httpapi/reg.go:64	register PUT@/prom_remote_write to HTTP server
log:2024-08-05T16:05:32.798+0800	INFO	http	httpapi/reg.go:64	register POST@/prom_remote_write to HTTP server
log:2024-08-05T16:05:32.798+0800	INFO	usage_trace	usagetrace/trace.go:152	add reserved input "prom_remote_write"
log:2024-08-05T16:05:34.446+0800	INFO	inputs	inputs/inputs.go:361	starting input prom_remote_write ...
log:2024-08-05T16:05:34.447+0800	INFO	prom_remote_write	promremote/input.go:94	prom_remote_write input started...
log:2024-08-05T16:05:34.447+0800	INFO	inputs	inputs/inputs.go:365	input prom_remote_write exited, this maybe a input that only register a HTTP handle

```

**Comment by coanor** [comment]
Plz explain what data missing.

---

There was a bug fix on prom remote write.

Before version 1.29.1(included), prom remote write data will attach datakit's host-global-tags, this is a buggy implement, because these data not come from the running datakit, and should not attache a tag `host=<datakit's hostname>`.

If you really need these tags, plz add these tags in prom-remote-write.conf

**Comment by saintak** [comment]
Indeed, as you said, all host tag data has been lost. After I manually set the following, it became normal:
```
  ## custom tags
  [inputs.prom_remote_write.tags]
   host = "myserver-hostname"
```
I think this is a **Breaking Changes** which needs to be specifically noted and show how to handle it in the release note 
> Plz explain what data missing.
> 
> There was a [bug fix #2244](https://docs.guance.com/en/datakit/changelog/#cl-1.30.0) on prom remote write.
> 
> Before version 1.29.1(included), prom remote write data will attach datakit's host-global-tags, this is a buggy implement, because these data not come from the running datakit, and should not attache a tag like `host=<datakit's hostname>`.
> 
> If you really need these tags, plz add these tags in prom-remote-write.conf like this:
> 
> ```toml
>   ## custom tags
>   [inputs.prom_remote_write.tags]
>   # some_tag = "some_value"
>   # more_tag = "some_other_value"
> ```

**Comment by coanor** [comment]
> Indeed, as you said, all host tag data has been lost. After I manually set the following, it became normal:
> 
> ```
>   ## custom tags
>   [inputs.prom_remote_write.tags]
>    host = "myserver-hostname"
> ```
> 
> I think this is a **Breaking Changes** which needs to be specifically noted and show how to handle it in the release note
> 
> > Plz explain what data missing.
> > There was a [bug fix #2244](https://docs.guance.com/en/datakit/changelog/#cl-1.30.0) on prom remote write.
> > Before version 1.29.1(included), prom remote write data will attach datakit's host-global-tags, this is a buggy implement, because these data not come from the running datakit, and should not attache a tag like `host=<datakit's hostname>`.
> > If you really need these tags, plz add these tags in prom-remote-write.conf like this:
> > ```toml
> >   ## custom tags
> >   [inputs.prom_remote_write.tags]
> >   # some_tag = "some_value"
> >   # more_tag = "some_other_value"
> > ```

It's maybe a break change(indeed got some *new feature*), but this is a bug introduced since version 1.25.0, not a expected new feature, so we made it a bug-fix, not a breaking change.

**Comment by saintak** [comment]
I think this is not only a 1.25.0 break change.
I find that the 1.25.0 released at (2024/03/06), yet I have use guance.com for prom_remote_write years' time, not since 2024/03/06.
I am sure that I use this without custom tags for a long time and no error before
